ouroboros-consensus/src/Ouroboros/Consensus/Storage/LedgerDB/InMemory.hs

```diff
@@ -355,18 +352,18 @@ ledgerDbFromGenesis params = ledgerDbWithAnchor params . genesisChainSummary
 type ResolveBlock m r b = r -> m b
 
 -- | Annotated ledger errors
-data AnnLedgerError l r e = AnnLedgerError {
+data AnnLedgerError l r = AnnLedgerError {
       -- | The ledger DB just /before/ this block was applied
       annLedgerState  :: LedgerDB l r
 
       -- | Reference to the block that had the error
     , annLedgerErrRef :: r
 
       -- | The ledger error itself
-    , annLedgerErr    :: e
+    , annLedgerErr    :: LedgerErr l
     }
 
-class Monad m => ResolvesBlocks r b m where
+class Monad m => ResolvesBlocks r b m | m -> b where
```

Comment:
try `r -> b`

Reply:
Ah, no, if I do that, even the base instance won't work anymore:

```haskell
instance Monad m => ResolvesBlocks r b (ReaderT (ResolveBlock m r b) m) where
  resolveBlock r = ReaderT $ \f -> f r
```

which makes sense, this does not impose any restrictions on the relation between `r` and `b`.